apps/backend/src/middlewares/maxVotesCheckFor.ts

```diff
@@ -0,0 +1,171 @@
+import { db } from '@repo/db';
+import type { BallotPaper as DBBallotPaper } from '@repo/db/types';
+import {
+  insertableBallotPaperSectionObject,
+  response400Object,
+  response404Object,
+  updateableBallotPaperObject,
+  updateableBallotPaperSectionObject,
+  zodErrorToResponse400,
+  type BallotPaper,
+  type BallotPaperSection,
+  type Election,
+  type Response400,
+} from '@repo/votura-validators';
+import type { NextFunction, Request, Response } from 'express';
+import { HttpStatusCode } from '../httpStatusCode.js';
+
+export enum RequestTypeMaxVotesCheck {
+  ballotPaperUpdate = 'ballotPaperUpdate',
+  ballotPaperSectionCreate = 'ballotPaperSectionCreate',
+  ballotPaperSectionUpdate = 'ballotPaperSectionUpdate',
+}
+
+/**
+ * Parses the request body to extract the max votes value based on the request type.
+ *
+ * @param reqType The type of request being made, which determines how to parse the body.
+ * @param req The request object containing the body to parse.
+ * @param res The response object to send errors to if parsing fails.
+ * @returns A promise that resolves to the max votes value if parsing is successful, or null if it fails.
+ */
+const parseRequestMaxVotes = async (
+  reqType: RequestTypeMaxVotesCheck,
+  req: Request,
+  res: Response<Response400>,
+): Promise<number | null> => {
+  let result = null;
+  if (reqType === RequestTypeMaxVotesCheck.ballotPaperUpdate) {
+    result = await updateableBallotPaperObject.safeParseAsync(req.body);
+  } else if (reqType === RequestTypeMaxVotesCheck.ballotPaperSectionCreate) {
+    result = await insertableBallotPaperSectionObject.safeParseAsync(req.body);
+  } else {
+    result = await updateableBallotPaperSectionObject.safeParseAsync(req.body);
+  }
+  const { data, error, success } = result;
+  if (!success) {
+    res.status(HttpStatusCode.badRequest).send(zodErrorToResponse400(error));
+    return null;
+  }
+  return data.maxVotes;
+};
+
+/**
+ * Get the maxVotes for the ballot paper with the given ballotPaperId
+ *
+ * @param ballotPaperId The id of the ballot paper to get maxVotes from
+ * @param res The response object to send errors to if getting the ballot paper fails
+ * @returns A promise that resolves to the max votes value if successful, or null if unsuccessful
+ */
+const getBallotPaperMaxVotes = async (
+  ballotPaperId: BallotPaper['id'],
+  res: Response<Response400>,
+): Promise<DBBallotPaper['maxVotes'] | null> => {
+  const ballotPaper = await db
+    .selectFrom('ballotPaper')
+    .select(['maxVotes'])
+    .where('id', '=', ballotPaperId)
+    .executeTakeFirst();
+
+  if (ballotPaper === undefined) {
+    res.status(HttpStatusCode.notFound).json(
+      response404Object.parse({
+        message: `Ballot paper not found.`,
+      }),
+    );
+    return null;
+  }
```

Comment:
```suggestion
): Promise<DBBallotPaper['maxVotes']> => {
  const ballotPaper = await db
    .selectFrom('ballotPaper')
    .select(['maxVotes'])
    .where('id', '=', ballotPaperId)
    .executeTakeFirstOrThrow();
```

Already check in `defaultBallotPaperChecks`